packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -859,15 +859,22 @@ class IntelliJValidatorOnMac extends IntelliJValidator {
 
   @override
   String get pluginsPath {
-    final List<String> split = version.split('.');
-    final String major = split[0];
-    final String minor = split[1];
-    return globals.fs.path.join(
-      globals.fsUtils.homeDirPath,
-      'Library',
-      'Application Support',
-      '$id$major.$minor',
-    );
+    final String plistFile = globals.fs.path.join(installPath, 'Contents', 'Info.plist');
+    final String altLocation = PlistParser.instance.getValueFromFile(plistFile, 'JetBrainsToolboxApp');
+
+    if (altLocation == null) {
+      final List<String> split = version.split('.');
```

Comment:
It looks like `version` can be `'unknown'`, which will cause `split[1]` to throw an index out of bounds exception. Instead, the value of `pluginsPath` should indicate that there was an error (maybe `null`), and the code in `IntelliJValidator.validate()` should check for that error, and modify the `ValidationResult` to signal the issue to the user.